src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,690 @@
+					continue;
+				}
+
+				ProcessAssembly (assembly, iterator);
+			}
+		}
+
+		protected virtual void ProcessAssembly (AssemblyDefinition assembly, XPathNodeIterator iterator)
+		{
+#if !FEATURE_ILLINK
+			if (IsExcluded (iterator.Current))
+				return;
+#endif
+			ProcessTypes (assembly, iterator.Current.SelectChildren ("type", _ns));
+		}
+
+		ArrayBuilder<Attribute> ProcessAttributes (XPathNodeIterator iterator)
+		{
+			var attributes = new ArrayBuilder<Attribute> ();
+			while (iterator.MoveNext ()) {
+				string attributeName = GetFullName (iterator.Current);
+				ArrayBuilder<string> arguments = GetAttributeArguments (iterator.Current.SelectChildren ("argument", _ns));
+				ArrayBuilder<string> fields = GetAttributeFields (iterator.Current.SelectChildren ("field", _ns));
+				ArrayBuilder<string> properties = GetAttributeProperties (iterator.Current.SelectChildren ("property", _ns));
+
+				attributes.Add (new Attribute (attributeName, arguments, fields, properties));
+			}
+			return attributes;
+		}
+
+		ArrayBuilder<string> GetAttributeArguments (XPathNodeIterator iterator)
+		{
+			ArrayBuilder<string> arguments = new ArrayBuilder<string> ();
+			while (iterator.MoveNext ()) {
+				arguments.Add (iterator.Current.Value);
+			}
+			return arguments;
+		}
+
+		ArrayBuilder<string> GetAttributeFields (XPathNodeIterator iterator)
+		{
+			ArrayBuilder<string> fields = new ArrayBuilder<string> ();
+			while (iterator.MoveNext ()) {
+				fields.Add (iterator.Current.Value);
+			}
+			return fields;
+		}
+
+		ArrayBuilder<string> GetAttributeProperties (XPathNodeIterator iterator)
+		{
+			ArrayBuilder<string> properties = new ArrayBuilder<string> ();
+			while (iterator.MoveNext ()) {
+				properties.Add (iterator.Current.Value);
+			}
+			return properties;
+		}
+
+		void ProcessTypes (AssemblyDefinition assembly, XPathNodeIterator iterator)
+		{
+			while (iterator.MoveNext ()) {
+				XPathNavigator nav = iterator.Current;
+
+				string fullname = GetFullName (nav);
+
+				if (IsTypePattern (fullname)) {
+					ProcessTypePattern (fullname, assembly, nav);
+					continue;
+				}
+
+				TypeDefinition type = assembly.MainModule.GetType (fullname);
+
+				if (type == null) {
+					if (assembly.MainModule.HasExportedTypes) {
+						foreach (var exported in assembly.MainModule.ExportedTypes) {
+							if (fullname == exported.FullName) {
+								var resolvedExternal = exported.Resolve ();
+								if (resolvedExternal != null) {
+									type = resolvedExternal;
+									break;
+								}
+							}
+						}
+					}
+				}
+
+				if (type == null)
+					continue;
+
+				ProcessType (type, nav);
+			}
+		}
+
+		static bool IsTypePattern (string fullname)
+		{
+			return fullname.IndexOf ("*") != -1;
+		}
+
+		static Regex CreateRegexFromPattern (string pattern)
+		{
+			return new Regex (pattern.Replace (".", @"\.").Replace ("*", "(.*)"));
```

Comment:
I can't find discussion that motivated adding support for `*` wildcard - did someone ask for this?

Reply:
I don't think there was a request commit is from 13 years ago https://github.com/mono/linker/commit/a8d7006e829e505c542a2cb77b350b3e208e2d9b